src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -150,74 +107,22 @@ def get_used_budget_for_project(self, project_id: int) -> float:
-
-        Returns a list of project IDs.
-
-        This may throw an exception if there is some sort of issue fetching the list from the redis
-        store.
-        """
-        return {int(project_id) for project_id in self.cluster.smembers(LPQ_MEMBERS_KEY)}
-
-    def is_lpq_project(self, project_id: int) -> bool:
-        """
-        Checks whether the given project is currently using the low priority queue.
-        """
-        return bool(self.cluster.sismember(LPQ_MEMBERS_KEY, project_id))
-
-    def add_project_to_lpq(self, project_id: int) -> bool:
-        """
-        Assigns a project to the low priority queue.
-
-        This registers an intent to redirect all symbolication events triggered by the specified
-        project to be redirected to the low priority queue.
-
-        Applies a backoff timer to the project which prevents it from being automatically evicted
-        from the queue while that timer is active.
-
-        Returns True if the project was a new addition to the list. Returns False if it was already
-        assigned to the low priority queue. This may throw an exception if there is some sort of
-        issue registering the project with the queue.
-        """
-
-        if self._is_backing_off(project_id):
-            return False
-        # If this successfully completes then the project is expected to be in the set.
-        was_added = int(self.cluster.sadd(LPQ_MEMBERS_KEY, project_id)) > 0
-        self._register_backoffs([project_id])
-        return was_added
-
-    def remove_projects_from_lpq(self, project_ids: set[int]) -> int:
-        """
-        Unassigns projects from the low priority queue.
-
-        This registers an intent to restore all specified projects back to the regular queue.
-
-        Applies a backoff timer to the project which prevents it from being automatically assigned
-        to the queue while that timer is active.
-
-        Returns the number of projects that were actively removed from the queue. Any projects that
-        were not assigned to the low priority queue to begin with will be omitted from the return
-        value. This may throw an exception if there is some sort of issue deregistering the projects
-        from the queue.
-        """
-        removable = [project for project in project_ids if not self._is_backing_off(project)]
+        average_used = total_sum / total_time_window / 1000
 
-        if not removable:
-            return 0
+        new_is_lpq = average_used > settings.SENTRY_LPQ_OPTIONS["project_budget"]
 
-        # This returns the number of projects removed, and throws an exception if there's a problem.
-        # If this successfully completes then the projects are expected to no longer be in the set.
-        removed = int(self.cluster.srem(LPQ_MEMBERS_KEY, *removable))
-        self._register_backoffs(removable)
-        return removed
+        self.cluster.set(name=member_key, value=int(new_is_lpq), ex=self._backoff_timer)
```

Comment:
Can you move that behind a `if new_is_lpq`?

Otherwise the initial trigger can only happen in `backoff_timer` intervals.
IMO it makes sense to have a "when triggered, project will remain in LPQ for at least `backoff_timer`".

The other way around makes less sense IMO.

Reply:
This differs slightly in behavior from how it was before. Before, when a project got taken off the LPQ, it couldn't go back on for `backoff_timer`.

Reply:
yes I know, though I challenge that this behavior makes sense. and I don’t think its implementable with a simple redis `TTL`.